packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -3670,7 +3670,7 @@ void main() {
 
     // The ListView has scrolled to keep the TextField and cursor handle
     // visible.
-    expect(scrollController.offset, 26.0);
+    expect(scrollController.offset, 28.0);
```

Comment:
Why do all these values need to change by 2 pixels?

Reply:
So this got me slightly puzzled. My semi-educated guess is that this is because in _FloatingCursorPainter.paintRegularCursor() there's two things going on:

1. Small per-platform adjustments:
```
    final double? caretHeight = renderEditable._textPainter.getFullHeightForCaret(textPosition, caretPrototype);
[...]
          caretRect = Rect.fromLTWH(
```

2. then, a bit of shifting:
```
    caretRect = caretRect.shift(renderEditable._paintOffset);
    final Rect integralRect = caretRect.shift(renderEditable._snapToPhysicalPixel(caretRect.topLeft));
[...]
    caretPaintCallback(integralRect);
```

It's probably because of 1. - not sure why this was only done in the paint routine and not in getLocalRectForCaret(). Maybe a topic for another PR to clean it up.

Reply:
This seems to cause some internal golden images to show up differently too. Maybe replace the code in `getLocalRectForCaret` with the implementation here for now?